src/client/containers/BoardPageComponents/CandidateList/CandidateListPreDeadline.js

```diff
@@ -0,0 +1,69 @@
+import React, { useState } from 'react';
+import { useCandidates } from '../../UseHooks/useCandidates';
+import { useUpdateBallots } from './useUpdateBallots';
+import { CardItemDecorator } from '../../../components/CandidateCard/CandidateCardItem.component';
+import { candidateCardSorting } from '../../../components/CandidateCard/CandidateCardSorting';
+import {
+  DragAndSortAdapter,
+  SortableItem,
+} from '../../DragAndSortAdapter/DragAndSortAdapter';
+import { onDragEnd } from '../../DragAndSortAdapter/OnDragEnd';
+import PropTypes from 'prop-types';
+import { deleteCandidate } from './deleteCandidate';
+import { useBoard } from '../../BoardPage/BoardProvider';
+
+export default function CandidateListPreDeadline({ userId, boardId }) {
+  const { candidates, setCandidates, error } = useCandidates(userId, boardId);
+  const [draggedInit, setDraggedInit] = useState(false);
+  const { setBoardLoading } = useBoard();
+
+  function candidateTransform(candidate, index) {
+    return {
+      ...candidate,
+      rank: index + 1,
+    };
+  }
+
+  useUpdateBallots(userId, boardId, candidates, draggedInit);
+
+  const handleDelete = (candidateId) => {
+    deleteCandidate(userId, boardId, candidateId);
```

Comment:
The delete ought to be awaited so the reload happens after

Reply:
I added setBoardLoading to true after it. Wouldn't it work?

Reply:
But what if the fetching happens before the delete then?

Reply:
Hmm, I'm not sure how I should solve it.

Reply:
Then I guess I have to send setBoardLoading with props to deleteCandidate and set it there?

Reply:
const handleDelete = async (candidateId) => {
    await eleteCandidate(userId, boardId, candidateId); // you need some error handling here
    setBoardLoading(true);
  };

Reply:
You need to add error handling to your request also. Lets defer that for now. Create an issue for it so you don't forget

Reply:
![image](https://user-images.githubusercontent.com/75583909/130676415-fe170424-f0b3-4ae9-b2b2-a79d1f5f4bee.png)

Reply:
deleteCandidate function was already async/await, why do we have to add it here too? Just trying to understand.

Reply:
Not from where you called it. 
Promises return immediately, remember
When a promise is awaited it returns from its context and recreates that same context from memory when it is resolved and then continues.